dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserSettingService.java

```diff
@@ -177,14 +178,14 @@ public Serializable getUserSetting(UserSettingKey key, String username) {
   @Transactional(readOnly = true)
   public Map<String, Serializable> getUserSettingsWithFallbackByUserAsMap(
       User user, Set<UserSettingKey> userSettingKeys, boolean useFallback) {
-    Map<String, Serializable> result =
-        Sets.newHashSet(getUserSettings(user)).stream()
-            .filter(
-                userSetting ->
-                    userSetting != null
-                        && userSetting.getName() != null
-                        && userSetting.getValue() != null)
-            .collect(Collectors.toMap(UserSetting::getName, UserSetting::getValue));
+    Map<String, Serializable> result = new HashMap<>();
+    Sets.newHashSet(getUserSettings(user)).stream()
+        .filter(
+            userSetting ->
+                userSetting != null
+                    && userSetting.getName() != null
+                    && userSetting.getValue() != null)
+        .forEach(userSetting -> result.put(userSetting.getName(), userSetting.getValue()));
```

Comment:
This is to make sure `getUserSettingsWithFallbackByUserAsMap()` never return `null`